go-controller/pkg/ovn/controller/services/load_balancer.go

```diff
@@ -293,11 +303,17 @@ func buildPerNodeLBs(service *v1.Service, configs []lbConfig, nodes []nodeInfo)
 
 				if config.externalTrafficLocal {
 					// for ExternalTrafficPolicy=Local, remove non-local endpoints from the router/switch targets
+					// NOTE: on the switches, filtered eps are used only by masqueradeVIP
 					routerV4targetips = util.FilterIPsSlice(routerV4targetips, node.nodeSubnets(), true)
 					routerV6targetips = util.FilterIPsSlice(routerV6targetips, node.nodeSubnets(), true)
 					switchV4targetips = util.FilterIPsSlice(switchV4targetips, node.nodeSubnets(), true)
 					switchV6targetips = util.FilterIPsSlice(switchV6targetips, node.nodeSubnets(), true)
 				}
+				if config.internalTrafficLocal {
```

Comment:
do u want make sure !ETPLocal too before u remove the none local eps?

Reply:
I think no right? So ITP and ETP are unrelated. Even if ETP=local, I want ITP=local to have its targetIPs filtered, but I get your point since if ETP=local then it would already be taken care of in L309/310. I thought its better to leave it as is to give code clarity. LMK if you feel adding !ETPLocal makes it better...

Reply:
this section from the doc what trigger my comment
`
Service Internal Traffic Policy is not used when externalTrafficPolicy is set to Local on a Service. It is possible to use both features in the same cluster on different Services, just not on the same Service.
`

Reply:
ah yeah hopefully https://github.com/kubernetes/website/pull/32877 will take care of that confusion.